File: src/main/LeetCode/python/maximum-binary-string-after-change.py
class MaximumBinaryStringAfterChange:
    def maximumBinaryString(self, binary: str) -> str:
        """
        https://leetcode.com/problems/maximum-binary-string-after-change/
        忽略前置'1', 统计'0'的数量即可

        :param binary:
        :return:
        """
        # 以下代码一行也可实现，但可读性差
        # head = binary.find('0')
        # if head == -1:
        #     return binary
        # else:
        #     count = binary.count('0')
        #     return '1' * (head + count - 1) + '0' + '1' * (len(binary) - head - count)

        # 更好的写法
        if '0' not in binary:
            return binary
        else:
            # 统计的是首位 0 后面 1 的数量
            count, length = binary.count('1', binary.find('0')), len(binary)
            return '1' * (length - count - 1) + '0' + '1' * count

File: src/main/LeetCode/java/MaximumBinaryStringAfterChange.java
package LeetCode.java;

import java.util.Arrays;

/**
 * 题目描述：https://leetcode.com/problems/maximum-binary-string-after-change/
 *
 * 思路简述：忽略前置'1', 统计'0'的数量即可
 */
public class MaximumBinaryStringAfterChange {
    public String maximumBinaryString(String binary) {
        int head = binary.indexOf('0');
        if (head == -1) {
            return binary;
        } else {
            int count = 0;
            for (int i = 0; i < binary.length(); i++) {
                count += binary.charAt(i) == '0' ? 1 : 0;
            }
            char[] chars = new char[binary.length()];
            Arrays.fill(chars, '1');
            chars[head + count - 1] = '0';
            return new String(chars);
        }
    }
}
